node/src/Pos/Update/Download.hs

```diff
@@ -106,36 +110,43 @@ downloadUpdateDo cps@ConfirmedProposalState {..} = do
             throwError "There's unapplied update already downloaded"
 
         logInfo "Downloading update..."
-        file <- ExceptT $ liftIO (downloadHash updateServers updHash) <&>
+        file <- ExceptT $ downloadHash updateServers updHash <&>
                 first (sformat ("Update download (hash "%build%
                                 ") has failed: "%stext) updHash)
 
+        logInfo $ "Update was downloaded, saving to " <> show updPath
+
         liftIO $ BSL.writeFile updPath file
-        logInfo "Update was downloaded"
+        logInfo $ "Update was downloaded, saved to " <> show updPath
         downloadedMVar <- views (lensOf @UpdateContext) ucDownloadedUpdate
         putMVar downloadedMVar cps
         logInfo "Update MVar filled, wallet is notified"
 
-    whenLeft res logWarning
+    whenLeft res logDownloadError
   where
+    handleErr = return . Left . show
```

Comment:
If you catch `SomeException`, you should use `reportOrLog` function in your handler to report fatal errors.